test/models/solution_test.rb

```diff
@@ -127,6 +127,112 @@ class SolutionTest < ActiveSupport::TestCase
     assert solution.exercise_solution_files["log_line_parser.rb"].start_with?("module LogLineParser")
   end
 
+  test "read_file for concept exercise returns exercise file" do
```

Comment:
For now, I opted to add the tests to this file instead of separate test files for the generate classes. I think that if we have the files controller tests verify that the right generated file is returned, it is fine to move these tests.

Reply:
Yeah - I'm ok with this for now. My preference would be to probably duplicate the relevant tests to each class, just so we have command-level test-coverage in case we ever access them from anywhere else, or add exceptions for something etc.

Reply:
Done

Reply:
Nice. Thanks :)